### PR TITLE
[K32W0] Read General Diagnostics cluster attribute at runtime

### PR DESCRIPTION
**Pull request by doru91**
#### Problem
What is being fixed? 
*  General Diagnostics cluster implementation was missing on K32W0 platform

#### Change overview
Add General Diagnostics cluster implementation on K32W0

#### Testing
How was this tested?
* manually tested using chip-tool

### CODE REVIEW

File: src/platform/nxp/k32w/k32w0/PlatformManagerImpl.cpp [hunk 1]
```diff
@@ -74,6 +79,108 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
     return err;
 }
 
+CHIP_ERROR PlatformManagerImpl::_Shutdown()
```
Comment (Damian-Nordic) [inline src/platform/nxp/k32w/k32w0/PlatformManagerImpl.cpp:82]: All platforms seem to be copy-pasting the same code for these attributes. Would it make sense to put it in GenericPlatformManagerImpl and share across all platforms?

**Comment by doru91** [comment]
This PR also needs to be cherry-picked to the master branch.

File: src/platform/nxp/k32w/k32w0/PlatformManagerImpl.cpp (hunk 1 above)
Comment (doru91) [reply src/platform/nxp/k32w/k32w0/PlatformManagerImpl.cpp:82]: The refactorization that you are proposing seems to be already present on the master branch. I would say that bringing that functionality to TE7 LTS support branch is not necessary. For the master branch I opened https://github.com/project-chip/connectedhomeip/pull/12468.

**Comment by doru91** [comment]
> This PR also needs to be cherry-picked to the master branch.

> /rebase

rebased

**Pull request closed without merging**